Sdk/Track/Swrve.h

```diff
@@ -354,6 +354,59 @@ typedef void (^SwrveResourcesUpdatedListener) ();
+ *
+ * The default user ID is a random UUID. The userID is cached in the
+ * default settings of the app and recalled the next time you initialize the
+ * app. This means the ID for the user will stay consistent for as long as the
+ * user has your app installed on the device.
+ *
+ * \param swrveAppID The App ID for your app supplied by Swrve.
+ * \param swrveAPIKey The secret token for your app supplied by Swrve.
+ * \param launchOptions The Application's launchOptions from didFinishLaunchingWithOptions.
+ * \returns An initialized Swrve object.
+ */
++(Swrve*) sharedInstanceWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey launchOptions:(NSDictionary*)launchOptions;
+
+/*! Creates and initializes the shared Swrve singleton.
+ * The userID is used by Swrve to identify unique users. It must be unique for all users
+ * of your app. The default user ID is a random UUID.
+ *
+ * \param swrveAppID The App ID for your app supplied by Swrve.
+ * \param swrveAPIKey The secret token for your app supplied by Swrve.
+ * \param swrveUserID The unique user id for your application.
+ * \param launchOptions The Application's launchOptions from didFinishLaunchingWithOptions.
+ * \returns An initialized Swrve object.
+ */
++(Swrve*) sharedInstanceWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey userID:(NSString*)swrveUserID launchOptions:(NSDictionary*)launchOptions;
+
+/*! Creates and initializes the shared Swrve singleton.
+ *
+ * Takes a SwrveConfig object that can be used to change default settings.
+ *
+ * \param swrveAppID The App ID for your app supplied by Swrve.
+ * \param swrveAPIKey The secret token for your app supplied by Swrve.
+ * \param swrveConfig The swrve configuration object used to override default settings.
+ * \param launchOptions The Application's launchOptions from didFinishLaunchingWithOptions.
+ * \returns An initialized Swrve object.
+ */
++(Swrve*) sharedInstanceWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey config:(SwrveConfig*)swrveConfig launchOptions:(NSDictionary*)launchOptions;
+
+/*! Creates and initializes the shared Swrve singleton.
+ *
+ * Takes a SwrveConfig object that can be used to change default settings.
+ * The userID is used by Swrve to identify unique users. It must be unique for all users
+ * of your app. The default user ID is a random UUID.
+ *
+ * \param swrveAppID The App ID for your app supplied by Swrve.
+ * \param swrveAPIKey The secret token for your app supplied by Swrve.
+ * \param swrveConfig The swrve configuration object used to override default settings.
+ * \param swrveUserID The unique user id for your application.
+ * \param launchOptions The Application's launchOptions from didFinishLaunchingWithOptions.
+ * \returns An initialized Swrve object.
+ */
++(Swrve*) sharedInstanceWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey userID:(NSString*)swrveUserID config:(SwrveConfig*)swrveConfig launchOptions:(NSDictionary*)launchOptions;
+
```

Comment:
Indeed, should we move userId into the config and leave the launchOptions as non optional there?